### PR TITLE
Primer docs nav restructuring

### PR DESCRIPTION
**Pull request by emilybrick**
This work is part of https://github.com/github/primer/issues/1307 and allows us to begin adding content to primer.style/design from surrounding docs repos (see epic for the full content migration task list). 

### Context
[Primer IA (ADR)](https://github.com/github/primer/blob/2023-01-10-primer-docs-IA/adrs/2023-01-10-primer-docs-IA.md)
[Primer IA spreadsheet](https://docs.google.com/spreadsheets/d/1MzdERnNsCeh_DAd5svKFBG9ojK9FDX4lgTkjCZjW2pI/edit#gid=0)
[Primer IA FigJam](https://www.figma.com/file/811c7oT909JpbnlLe7biXn/Primer-docs-IA?node-id=0%3A1&t=fjAasShNy8MZ4Ax2-1)

### Tasks
- [x] update doctocat version to 4.3.0  ([Doctocat PR](https://github.com/primer/doctocat/pull/534))
  - _Note: This version allows NavList to next two levels deep, which is necessary to implement [the new Primer IA](https://github.com/github/primer/blob/2023-01-10-primer-docs-IA/adrs/2023-01-10-primer-docs-IA.md)_
- [x] Rename top level docs sections to match the new Primer docs IA 
  - _Note: I'm only editing the sections where content already exists, so sections like "Native platforms" aren't added yet._
- [x] ~Move /accessibility (and child pages) to /guides~ Not handling in this PR, will do as a separate chunk ([issue](https://github.com/github/primer/issues/1707))
- [x] Move /tools/figma to /guides 
- [x] Update index page for /guides

### CODE REVIEW

**Review by josepmartins — APPROVED** [review]
LGTM! ✨

File: src/@primer/gatsby-theme-doctocat/nav.yml [hunk 1]
```diff
@@ -1,12 +1,17 @@
-- title: Foundations
-  url: /foundations
+- title: Guides
+  url: /guides
```
[the remark this answers is not in the document]
Comment (emilybrick) [reply src/@primer/gatsby-theme-doctocat/nav.yml:2]: Yeah, I'll update that in nav.yml. I'm assuming folks can still hit the url from the address bar (e.g. primer.style/design/guides)? That way we can just make sure those pages have content in them, at least a list of the child pages

[the remark this answers is not in the document]
Comment (colebemis) [reply src/@primer/gatsby-theme-doctocat/nav.yml:2]: Yup 👍

**Review by colebemis — APPROVED** [review]
Looks good 🚀

**Pull request merged**